File: examples/example2.py
#!/usr/bin/python
#
# File: example2.py       I. Henry   March 30 2005
#
# Python translation of example2.c from the WFDB Programmer's Guide
#
# Copyright (C) 2005 Isaac C. Henry (ihenry@physionet.org)

import wfdb, sys

def main(argv):
    an = wfdb.WFDB_AnninfoArray(2)
    annot = wfdb.WFDB_Annotation()
    if len(argv) < 2:
        print "usage:", argv[0], "record"
        sys.exit(1)
    a = an[0]
    a.name = "atr"
    a.stat = wfdb.WFDB_READ
    an[0] = a
    a = an[1]
    a.name = "aha"
    a.stat = wfdb.WFDB_AHA_WRITE
    an[1] = a
    if wfdb.annopen(argv[1], an.cast(), 2) < 0: sys.exit(2)
    while 1:
        if not (wfdb.getann(0, annot) == 0 and wfdb.putann(0,annot) == 0):
            break
    wfdb.wfdbquit()
    
if __name__ == "__main__":
    main(sys.argv)

File: examples/example2.pl
#!/usr/bin/perl 
#
# File: example2.pl       I. Henry   March 29 2005
#
# Perl translation of example2.c from the WFDB Programmer's Guide
#				   
# Copyright (C) 2005 Isaac C. Henry (ihenry@physionet.org)

package wfdb;
use wfdb;

$an = new wfdb::WFDB_AnninfoArray(2);
$annot = new wfdb::WFDB_Annotation;
if ( @ARGV < 1 ) {
    print STDERR "usage: ", $0, " record\n";
    exit(1);
}
$a = $an->getitem(0);
$a->{name} = "atr"; $a->{stat} = $WFDB_READ;
$an->setitem(0,$a);
$a = $an->getitem(1);
$a->{name} = "aha"; $a->{stat} = $WFDB_AHA_WRITE; 
$an->setitem(1,$a);
exit(2) if (annopen($ARGV[0], $an->cast(), 2) < 0);
L: goto L while (getann(0, $annot) == 0 && putann(0, $annot) == 0);
wfdbquit();
